runtime-light/stdlib/time/timelib-functions.cpp

```diff
@@ -13,11 +13,55 @@
 
 #include "common/containers/final_action.h"
 #include "runtime-common/core/allocator/platform-malloc-interface.h"
+#include "runtime-light/allocator/allocator.h"
 #include "runtime-light/stdlib/diagnostics/logs.h"
 #include "runtime-light/stdlib/time/time-state.h"
 
 namespace kphp::timelib {
 
+namespace {
+
+template<typename Opt, typename Func>
+void apply_if_has_value(Opt&& opt, Func&& fn) noexcept(noexcept(fn(*opt))) {
```

Comment:
I think it's better to get rid of this helper. If you want to add some `FP` into the PR, you may use `std::optional`'s monadic operations

Reply:
It looks weird but ok